### PR TITLE
Made some change so that lifecycle related APIs can be AWS S3 compatible.

### LINKED ISSUE #893: S3 APIs are not AWS S3 compatible.
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
> /kind bug

**What happened**:
Two parts:
1. Currently, the URL of S3 APIs is like http(s)://ip:port/v1/s3/bucketname, which is not compatible with AWS S3, v1/s3/ should be removed.
At the same time, the authentification used a private algorithm, which is not compatible with AWS S3 too.
2. Each s3 API did not do AWS s3 compatilbe test, they may not be AWS S3 compatible.

Because of the above reasons, the third party s3 client, like the s3 browser, cannot be used to test multi-cloud.

**What you expected to happen**:
The third-party s3 client, like the s3 browser, cannot be used to test multi-cloud.

**How to reproduce it (as minimally and precisely as possible)**:
NA

**Anything else we need to know?**:

**Environment**:
- Gelato(release/branch) version:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by sfzeng**
**What this PR does / why we need it**:
This PR is to make sure lifecylce related APIs can be AWS S3 compatible. The following test cases are passed:
s3tests_boto3.functional.test_s3.test_lifecycle_set
s3tests_boto3.functional.test_s3.test_lifecycle_get
s3tests_boto3.functional.test_s3.test_lifecycle_get_no_id
s3tests_boto3.functional.test_s3.test_lifecycle_id_too_long
s3tests_boto3.functional.test_s3.test_lifecycle_same_id
s3tests_boto3.functional.test_s3.test_lifecycle_invalid_status
s3tests_boto3.functional.test_s3.test_lifecycle_expiration_header_head
s3tests_boto3.functional.test_s3.test_lifecycle_set_noncurrent
s3tests_boto3.functional.test_s3.test_lifecycle_set_multipart

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #893 

**Special notes for your reviewer**:
It shows there are more than six hundreds of new line, but actually most of them are from protobuf, which is automatically created by protoc.

**Release note**:

```release-note
1. Currently, for expiration rule, we only support expiration days, not support sepcific expiration date.
2. As Prefix in rule is deprecatd, and Filter is suggested for AWS S3 itself. So we only support Filter. And curretnly, we only support Filter:Prefix, not support Filter:Tag.
```

### CODE REVIEW

File: s3/pkg/service/object.go [hunk 1]
```diff
@@ -316,11 +318,38 @@ func (s *s3Service) GetObjectMeta(ctx context.Context, in *pb.Object, out *pb.Ge
...
+func getObjectExpDate(lcRules []*pb.LifecycleRule, objKey string, lastModified int64) (int64, string) {
+	var expTime int64 = 0
+	for _, r := range lcRules {
+		if r.Status == utils2.RuleStatusDisabled {
+			continue
+		}
+		for _, ac := range r.Actions {
+			if ac.Name == utils2.ActionNameExpiration {
+				if r.Filter.Prefix == "" || strings.HasPrefix(objKey, r.Filter.Prefix) {
+					exp := lastModified + 60*60*24*int64(ac.Days)
```
Comment (kumarashit) [inline s3/pkg/service/object.go]: Put 60   *  60   *   24 as constant

File: s3/pkg/service/object.go [hunk 2]
```diff
@@ -316,11 +318,38 @@ func (s *s3Service) GetObjectMeta(ctx context.Context, in *pb.Object, out *pb.Ge
...
+					if expTime < exp {
```
Comment (kumarashit) [inline s3/pkg/service/object.go]: When can it land in this situation? i.e. expTime < exp

Comment (sfzeng) [reply s3/pkg/service/object.go]: If there are more than one expiration rule, but expiration days are not the same, we will remove object when the latest expiration time arrived.

File: s3/pkg/service/object.go (hunk 1 above)
Comment (sfzeng) [reply s3/pkg/service/object.go]: Done

File: s3/pkg/service/object.go (hunk 2 above)
Comment (kumarashit) [reply s3/pkg/service/object.go]: Got it

**Pull request merged**